### PR TITLE
Merged home and my trees

### PR DESCRIPTION
**Pull request by mahmed0715**
From #839 partially done, 
Home and My trees are merged, Now /my-trees is obsolete , we must replace the url to /home in database..
@sagararyal Please advice.
Thanks

### CODE REVIEW

File: app/assets/index.js [hunk 1]
```diff
@@ -11,6 +11,10 @@ import heart_outline from './images/icons/heart_outline.png';
 import rocket_outline from './images/icons/rocket_outline.png';
 import shovel_outline from './images/icons/shovel_outline.png';
 import target_outline from './images/icons/target_outline.png';
+import calender_green from './images/icons/calender_green.png';
```
Comment (norbertschuler) [inline app/assets/index.js]: The white and green calendar and tree icons also have been added in https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1178/files (which I do not know when it will be merged) - we should unify this as soon as both PRs are merged to avoid having the images twice.

File: app/components/UserContributions/RecurringCard.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            Sagar Aryal
+          </div>
+          <div>
+            <img className="menu-icon icon" src={images.calender_green} />Renews
+            on March 7, 2019
+          </div>
+        </div>
+        <div>
+          <span>
+            <i className="material-icons">delete_outline</i>
```
Comment (norbertschuler) [inline app/components/UserContributions/RecurringCard.js]: This button is without any functionality, isn't it?

File: app/components/UserContributions/RecurringCard.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          </span>
+          <span>
+            <i className="material-icons">edit</i>
```
Comment (norbertschuler) [inline app/components/UserContributions/RecurringCard.js]: This button is without any functionality, isn't it?

File: app/components/UserContributions/RecurringCard.js (hunk 2 above)
Comment (norbertschuler) [reply app/components/UserContributions/RecurringCard.js]: On my staging test account the buttons do nothing for "Recurring Tree Donations", but the delete button works for "All Tree Contributions":

Bildschirmfoto 2019-08-09 um 12 35 46

File: app/components/PublicTreeCounter/TreecounterHeader.js [hunk 4]
```diff
@@ -16,11 +16,16 @@ const TreecounterHeader = ({
...
+    let c = text.split(' ');
+    if (c.length > 1) c.length--;
+    return c.join(' ');
+  };
   return (
     <div className="tree-counter-profile flex-column">
       <UserProfileImage profileImage={logo} />
       <div className="user-info">
-        <div className="tree-counter-name">{caption}</div>
+        <div className="tree-counter-name">{onlyFirstName(caption)}</div>
```
Comment (norbertschuler) [inline app/components/PublicTreeCounter/TreecounterHeader.js]: @sagararyal Is it really wanted? For normal names it's o.k., but what about names not using the form "firstname lastname", e.g. my TPO account is named "Norberts Baumschule" (=" Norbert's Treeschool") which looks wrong if cut to the first part:

Bildschirmfoto 2019-08-09 um 12 39 42

File: app/components/UserContributions/RecurringCard.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import { delimitNumbers } from '../../utils/utils';
+import moment from 'moment';
+import 'moment/min/locales';
+import i18n from '../../locales/i18n.js';
+import { getDateFromMySQL } from '../../helpers/utils';
+
+const RecurringCard = ({ cardData, deleteContribution }) => {
+  return (
+    <div className="card" key={cardData.id}>
+      <div className="currency">€ 150.00</div>
```
Comment (norbertschuler) [inline app/components/UserContributions/RecurringCard.js]: I think the file `RecurringCard.js` is work in progress having constant and untranslated resources and buttons without functions below.

File: app/components/UserHome/index.js [hunk 6]
```diff
@@ -78,44 +83,62 @@ export default class UserHome extends Component {
...
-              />
-            )}
+          <div className="column right app-container__sidenav--heading">
+            <div className="heading">Recurring Tree Donations</div>
+            <div className="m-b-20">
+              {[{ id: 1 }, { id: 2 }].map(cardData => (
+                <RecurringCard cardData={cardData} key={cardData.id} />
+              ))}
+            </div>
+            <div className="heading">All Tree Contributions</div>
```
Comment (norbertschuler) [inline app/components/UserHome/index.js]: Please add a resource for this.

File: app/components/UserContributions/ContributionCard.js [hunk 7]
```diff
@@ -49,61 +50,55 @@ export default class ContributionCard extends React.Component {
...
-    return (
-      i18n.t('label.planted_on', {
-        date: moment(getDateFromMySQL(plantDate)).format('DD MMM YYYY')
-      }) +
-      ', ' +
-      i18n.t('label.added_on', {
-        date: moment(getDateFromMySQL(registrationDate)).format('DD MMM YYYY')
-      })
-    );
+    return i18n.t('label.renews_on', {
```
Comment (norbertschuler) [inline app/components/UserContributions/ContributionCard.js]: I wonder if this change would make show all contributions and donations from the user show only a plantDate with the "Renewed on" label while some of them are no recurring donations but one time contributions et al where the old and now deleted code applies? Am I right or wrong here?

**Comment by norbertschuler** [comment]
Will we continue working on this PR?

**Comment by mahmed0715** [comment]
There are some works to do still.. but its good to merge at least, there are some works to do in Backend for this to work the left my-trees menu updated..

**Comment by norbertschuler** [comment]
> but its good to merge at least

I am not sure that this PR is ready to be merged. I made two comments about the "renew date" which I think should not always be shown and the new kind of showing a tree counters title.

**Comment by sagararyal** [comment]
@norbertschuler this is work in progress, should not be merged as there are still some issues in the conditions and the content.

File: app/components/PublicTreeCounter/TreecounterHeader.js (hunk 4 above)
Comment (sagararyal) [reply app/components/PublicTreeCounter/TreecounterHeader.js]: @norbertschuler No this has to be fixed. always show full name.

**Comment by mahmed0715** [comment]
Fixed #839

File: app/config/index.js.dist [hunk 8]
```diff
@@ -1,21 +0,0 @@
...
-};
-
-export const context = {
-  scheme: 'https', // API server protocol
-  host: 'test.trilliontreecampaign.org', // API server domain
-  base: '',  // API base url. Debug mode off: "" on: "/app_dev.php" (requires login)
-  debug: true, // local console debugging switch
-  currency: 'USD',
-  mapIds: { inventory: 'dee6acf9de774fe6878813f707b4ab88'}
-};
```
Comment (norbertschuler) [inline app/config/index.js.dist]: This file should not be deleted!

File: package-lock.json [hunk 9]
```diff
@@ -5,75 +5,80 @@
   "requires": true,
   "dependencies": {
     "@babel/code-frame": {
-      "version": "7.0.0",
```
Comment (norbertschuler) [inline package-lock.json]: Changes of the `package-lock.json` file shouldn't be part of this patch as `package.json` was not modified.

**Comment by norbertschuler** [comment]
Check https://github.com/Plant-for-the-Planet-org/treecounter-app/tree/feature/merged_home_my_forest for fixing the merge conflicts (can be merged into this branch if it's o.k.)

**Comment by norbertschuler** [comment]
Please compare the old and new list of trees to see where there are still differences in the display of the user contributions (be aware not all cases are mapped by these examples!).

Also not the full list is displayed as the gifts which are a different part in the API response for https://staging.trilliontreecampaign.org/api/v1.1/de/loadUserProfile  are not yet shown (your new list shows the `contributions` array of the API response, but not the `gifts` array which formally was both shown in the list of my-trees):

0 0 0 0_8080_my-trees

0 0 0 0_8080_home

File: app/config/index.js.dist (hunk 8 above)
Comment (norbertschuler) [reply app/config/index.js.dist]: I added this file again in my custom branch https://github.com/Plant-for-the-Planet-org/treecounter-app/tree/feature/merged_home_my_forest

File: package-lock.json (hunk 9 above)
Comment (norbertschuler) [reply package-lock.json]: I discarded the changes in my custom branch https://github.com/Plant-for-the-Planet-org/treecounter-app/tree/feature/merged_home_my_forest

**Comment by mahmed0715** [comment]
This PR is moved to feature/merged_home_my_forest-merged-develop branch

**Comment by mahmed0715** [comment]
@jmiridis need giver details in contributions.
